src/internet_identity/src/main.rs

```diff
@@ -786,6 +789,44 @@ fn http_request(req: HttpRequest) -> HttpResponse {
     }
 }
 
+/// List of recommended security headers as per https://owasp.org/www-project-secure-headers/
+/// These headers enable browser security features (like limit access to platform apis and set
+/// iFrame policies, etc.).
+fn security_headers() -> Vec<HeaderField> {
+    vec![
+        ("X-Frame-Options".to_string(), "DENY".to_string()),
+        ("X-Content-Type-Options".to_string(), "nosniff".to_string()),
+
+        // "Referrer-Policy: no-referrer" would be more strict, but breaks local dev deployment
+        // same-origin is still ok from a security perspective
+        ("Referrer-Policy".to_string(), "same-origin".to_string()),
+        (
+            "Permissions-Policy".to_string(),
+            "accelerometer=(),\
+             autoplay=(),\
+             camera=(),\
+             display-capture=(),\
+             document-domain=(),\
+             encrypted-media=(),\
+             fullscreen=(),\
+             geolocation=(),\
+             gyroscope=(),\
+             magnetometer=(),\
+             microphone=(),\
+             midi=(),\
+             payment=(),\
+             picture-in-picture=(),\
+             publickey-credentials-get=(self),\
+             screen-wake-lock=(),\
+             sync-xhr=(self),\
+             usb=(),\
+             web-share=(),\
+             xr-spatial-tracking=()"
+                .to_string(),
+        ),
```

Comment:
Thanks for adding this! Generally looks good. I saw that some features are missing in the list. Btw. [permissionspolicy.com](https://www.permissionspolicy.com/) is a nice tool to generate these. 

If I disable everything, also proposed and experimental features, I get: 
```
Permissions-Policy: accelerometer=(), ambient-light-sensor=(), autoplay=(), battery=(), camera=(), cross-origin-isolated=(), display-capture=(), document-domain=(), encrypted-media=(), execution-while-not-rendered=(), execution-while-out-of-viewport=(), fullscreen=(), geolocation=(), gyroscope=(), keyboard-map=(), magnetometer=(), microphone=(), midi=(), navigation-override=(), payment=(), picture-in-picture=(), publickey-credentials-get=(self), screen-wake-lock=(), sync-xhr=(self), usb=(), web-share=(), xr-spatial-tracking=(), clipboard-read=(), clipboard-write=(), gamepad=(), speaker-selection=(), conversion-measurement=(), focus-without-user-activation=(), hid=(), idle-detection=(), interest-cohort=(), serial=(), sync-script=(), trust-token-redemption=(), window-placement=(), vertical-scroll=()
```
WDYT, would it make sense to extend it?

Reply:
Yes, thanks.